wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -308,7 +312,25 @@ public boolean isWorkunitComplete(String wuid) throws Exception
      */
     static public boolean isWorkunitComplete(WorkunitInfo thewui)
     {
-        return isWorkunitComplete(getStateID(thewui));
+        WUActionCode origAction = WUActionCode.fromName(thewui.getActionEx()); // some WUs have ActionEX as a string
```

Comment:
@drealeed please take a look at this change, as it might have an impact on your code...